tools/jenkins-slave-creation-unix/README.md

```diff
@@ -15,4 +15,66 @@
 <!--- specific language governing permissions and limitations -->
 <!--- under the License. -->
 
-This Terraform setup will spawn an instance that is ready to be saved into an AMI to create a Jenkins slave.
\ No newline at end of file
+This Terraform setup will spawn an instance that is ready to be saved into an AMI to create a Jenkins slave.
+
+# Steps
+## Setup Terraform
+### Fetch Terraform and unzip the binary
+
+```
+wget https://releases.hashicorp.com/terraform/0.12.24/terraform_0.12.24_linux_amd64.zip
+sudo apt install unzip
+unzip terraform_0.12.24_linux_amd64.zip
+```
+
+### Add to path
+Add the binary to the environment variable 'PATH'. 
+For example
+
+```
+sudo mv terraform /usr/local/bin/
+mkdir /home/ubuntu/bin
+mv /usr/local/bin/terraform /home/ubuntu/bin/terraform
+```
+
+### Verify 
+Check whether the terraform binary is in the PATH variable
+
+```
+echo $PATH
+```
+
+Verify terraform is properly installed
+
+```
+$ terraform --version
+Terraform v0.12.24
+$ which terraform
+/home/ubuntu/bin/terraform
+```
+
+## Python package requirements
+Install the terraform python package
+
+```
+pip3 install python_terraform
+```
+
+## Fill the redacted information
+- infrastructure.tf [Security groups]
+- infrastructure.tfvars [`key_name`, `key_path`, `secret_manager_docker_hub_arn`]
+- `~/.aws/config` [Isengard account profile]
+
+## Run the AMI creation script
+
+```
+./create_slave.sh
+```
+
+- Enter the desired directory
+
+## Create an AMI
+- Login to AWS Console
+- Instance would be created with the name used in `infrastructure.tfvars.instance_name`
+- Stop the instance & wait till the status changes from Running -> Stopping -> Stopped
```

Comment:
Do NOT stop the instance! 

Don't you remember the last time you stopped the instance while the Ami was being generated and the image turned out corrupt?

The system automatically stops the instance when it's done. Just wait for it.

Reply:
Check `/var/log/cloud-init-output.log` to find the error why instance didn't stop

Reply:
Good catch! Thanks for pointing out.